src/System.Console/src/System/TermInfo.cs

```diff
@@ -119,10 +119,20 @@ private Database(string term, byte[] data)
                 _term = term;
                 _data = data;
 
+                const int magicLegacyNumber = 0x11A; // magic number octal 0432 for legacy ncurses terminfo
+                const int magic32BitNumber = 0x21E; // magic number octal 01036 for new ncruses terminfo
+                bool reasAs32Bit = false; // default to legacy format
+
+                short magic = ReadInt16(data, 0);
```

Comment:
nit, this sequency is written wierdly to me, I would have expected
```
if (magic == magicLegacyNumber)
..
else if (magic == magic32BitNumber)
..
else
.. error
```
or a switch satement.

Reply:
also it might be simpler to just set up a delegate initialized to either ReadInt32 or ReadInt16, then use that delegate from then on. This eliminates the boolean `readAs32Bit` which makes the code easier to read.

Reply:
> this sequency is written wierdly to me

Not a big deal, but personally I'd write it as:
```C#
short readAs32Bit =
    magic == MagicLegacyNumber ? false :
    magic == Magic32BitNumber ? true :
    throw new InvalidOperationException(SR.IO_TermInfoInvalid);
```
This isn't an area that needs to be super optimized, so an extra branch for clarity is fine.